include/boost/geometry/algorithms/detail/buffer/buffer_policies.hpp

```diff
@@ -157,54 +151,116 @@ struct buffer_turn_info
         >
 {
     typedef Point point_type;
-    typedef RobustPoint robust_point_type;
 
-    std::size_t turn_index; // TODO: this might go if partition can operate on non-const input
+    std::size_t turn_index;
 
+    // Still necessary for turn-in-original calculations
     RobustPoint robust_point;
 
-    inline RobustPoint const& get_robust_point() const
-    {
-        return robust_point;
-    }
-
-    intersection_location_type location;
-
-    robust_point_type rob_pi, rob_pj, rob_qi, rob_qj;
-
-    std::size_t count_within;
+    // Information if turn can be used. It is not traversable if it is within
+    // another piece, or within the original (depending on deflation),
+    // or (for deflate) if there are not enough points to traverse it.
+    bool is_turn_traversable;
 
+    bool close_to_offset;
+    bool is_linear_end_point;
     bool within_original;
     signed_size_type count_in_original; // increased by +1 for in ext.ring, -1 for int.ring
 
-    std::size_t count_on_offsetted;
-    std::size_t count_on_helper;
-    std::size_t count_within_near_offsetted;
-
     inline buffer_turn_info()
         : turn_index(0)
-        , location(location_ok)
-        , count_within(0)
+        , is_turn_traversable(true)
+        , close_to_offset(false)
+        , is_linear_end_point(false)
         , within_original(false)
         , count_in_original(0)
-        , count_on_offsetted(0)
-        , count_on_helper(0)
-        , count_within_near_offsetted(0)
     {}
 };
 
-struct buffer_operation_less
+struct buffer_less
+{
+    template <typename Indexed>
+    inline bool operator()(Indexed const& left, Indexed const& right) const
+    {
+        if (! (left.subject->seg_id == right.subject->seg_id))
+        {
+            return left.subject->seg_id < right.subject->seg_id;
+        }
+
+        // Both left and right are located on the SAME segment.
+        if (! (left.subject->fraction == right.subject->fraction))
+        {
+            return left.subject->fraction < right.subject->fraction;
+        }
+
+        return left.turn_index < right.turn_index;
+    }
+};
+
+struct piece_get_box
+{
+    template <typename Box, typename Piece>
+    static inline void apply(Box& total, Piece const& piece)
+    {
+        if (piece.m_piece_border.m_has_envelope)
+        {
+            geometry::expand(total, piece.m_piece_border.m_envelope);
```

Comment:
ExpandBoxBox strategy is needed here.

Reply:
Thanks, will fix (this probably was already the case earlier, but will fix anyway)